subworkflows/local/call_repeat_expansions/main.nf

```diff
@@ -13,40 +14,60 @@ workflow CALL_REPEAT_EXPANSIONS {
     ch_trgt_bed // channel: [mandatory] [ val(meta), path(bed) ]
 
     main:
-    ch_repeat_calls_vcf = Channel.empty()
-    ch_versions         = Channel.empty()
+    ch_versions = Channel.empty()
 
     ch_bam_bai
         .map { meta, bam, bai -> [meta, bam, bai, meta.sex] }
         .set { ch_trgt_input }
 
     // Run TGRT
-    TRGT ( ch_trgt_input, ch_fasta, ch_fai, ch_trgt_bed.map { it[1] } )
+    TRGT (
+        ch_trgt_input,
+        ch_fasta,
+        ch_fai,
+        ch_trgt_bed
+    )
+    ch_versions = ch_versions.mix(TRGT.out.versions)
 
     // Sort and index bam
-    SAMTOOLS_SORT_TRGT ( TRGT.out.bam, [[],[]] )
-    SAMTOOLS_INDEX_TRGT(SAMTOOLS_SORT_TRGT.out.bam)
+    SAMTOOLS_SORT_TRGT (
+        TRGT.out.bam,
+        [[],[]]
+    )
+    ch_versions = ch_versions.mix(SAMTOOLS_SORT_TRGT.out.versions)
+
+    SAMTOOLS_INDEX_TRGT ( SAMTOOLS_SORT_TRGT.out.bam )
+    ch_versions = ch_versions.mix(SAMTOOLS_INDEX_TRGT.out.versions)
+
+    // Add FOUND_IN=TRGT tag
+    ADD_FOUND_IN_TAG (
+        TRGT.out.vcf.map { meta, vcf -> [ meta, vcf, [] ] },
+        "TRGT"
+    )
 
     // Sort and index bcf
-    BCFTOOLS_SORT_TRGT(TRGT.out.vcf)
+    BCFTOOLS_SORT_TRGT ( ADD_FOUND_IN_TAG.out.vcf )
+    ch_versions = ch_versions.mix(BCFTOOLS_SORT_TRGT.out.versions)
 
     BCFTOOLS_SORT_TRGT.out.vcf
         .join( BCFTOOLS_SORT_TRGT.out.tbi )
-        .map { meta, bcf, csi -> [ [ id : meta.project ], bcf, csi ] }
+        .map { meta, bcf, tbi -> [ [ id : meta.project ], bcf, tbi ] }
         .groupTuple()
         .set{ ch_bcftools_merge_in }
 
-    BCFTOOLS_MERGE ( ch_bcftools_merge_in, ch_fasta, ch_fai, [[],[]] )
-
-
-    ch_versions = ch_versions.mix(TRGT.out.versions)
-    ch_versions = ch_versions.mix(SAMTOOLS_SORT_TRGT.out.versions)
-    ch_versions = ch_versions.mix(SAMTOOLS_INDEX_TRGT.out.versions)
-    ch_versions = ch_versions.mix(BCFTOOLS_SORT_TRGT.out.versions)
+    BCFTOOLS_MERGE (
+        ch_bcftools_merge_in,
+        ch_fasta,
+        ch_fai,
+        [[],[]]
+    )
     ch_versions = ch_versions.mix(BCFTOOLS_MERGE.out.versions)
 
     emit:
-    vcf      = BCFTOOLS_SORT_TRGT.out.vcf  // channel: [ val(meta), path(vcf) ]
-    versions = ch_versions                 // channel: [ versions.yml ]
+    sample_vcf  = BCFTOOLS_SORT_TRGT.out.vcf  // channel: [ val(meta), path(vcf) ]
+    project_vcf = BCFTOOLS_MERGE.out.vcf      // channel: [ val(meta), path(vcf) ]
+    sample_bam  = SAMTOOLS_SORT_TRGT.out.bam  // channel: [ val(meta), path(bam) ]
+    sample_bai  = SAMTOOLS_INDEX_TRGT.out.bai // channel: [ val(meta), path(bai) ]
```

Comment:
If its a sample_bam with its corresponding sample_bai, wouldn't it be better to output them together as one channel?

Reply:
I think what I wrote in the previous comment holds true somewhat here as well, some tools that require a bam file do not require a bai (although to be fair most do). 

I think the tests get easier to read and write if we can standardise and know that there's only a meta and a file in the output channels from each workflow.

E.g.: 

```
path(workflow.out.project_vcf[0][1])).vcf.variantsMD5,
{ assert process.out.tbi[0][1].endsWith(".tbi") }
```

vs. 

```
path(workflow.out.project_vcf_tbi[0][0][0])).vcf.variantsMD5,
{ assert process.out.project_vcf_tbi[0][0][1].endsWith(".tbi") }
```

Especially if there's a mix of output channels with different number of elements (which there will be, because not all file types have an index). 

Most nf-core modules output one file per output, so to me it makes sense to try to keep subworkflows the same as well.